src/base/bittorrent/peerinfo.cpp

```diff
@@ -408,3 +408,8 @@ QString PeerInfo::flagsDescription() const
 {
     return m_flagsDescription;
 }
+
+int PeerInfo::downloadingPieceIndex() const
```

Comment:
Unfortunately, not everything looks good after some thinking. Here is the problem:

`TorrentInfo::filesForPiece()` is good function (and I need it for another similar feature, highlighting of the downloading files and listing them in the tooltip for the torrent progress bars, not submitted yet).  Thus should we implement `PeerInfo::downloadingFiles()` we either need to pass a reference to `TorrentInfo` instance there (currently it has `libtorrent::peer_info` or to duplicate code ( :-1: ). At the same time the current solution seems to be normalized (in terms of relational DB normalization). I guess the libtorrent authors arrived to the current design not unintentionally.